File: Assignment_1/WEB-INF/classes/NewCustomerPage.java
import javax.servlet.*;
import javax.servlet.http.*;
import java.util.*;
import java.io.*;
import java.net.*;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;

@WebServlet("/NewCustomerPage")

public class NewCustomerPage extends HttpServlet
{	
	protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException,IOException{
		response.setContentType("text/html");
		PrintWriter out =response.getWriter();
		Utilities util=new Utilities(request,out);
		if(request.getAttribute("checkMsg") != null)
		{
			out.println("<h3 style='color:black;background-color:blue;'>"+(String)(request.getAttribute("checkMsg"))+"</h3>");
			request.setAttribute("checkMsg","");
		}	
		util.printHtml("Header2.html");
		out.print("<div class='post' style='float: none; width: 100%'>");
		out.print("<h2 class='title meta'><a style='font-size: 24px;'>Add New Customer</a></h2>"
				+ "<div class='entry'>"
				+ "<div style='width:400px; margin:25px; margin-left: auto;margin-right: auto;'>");
		out.print("<form action=''>"
				+ "<table style='width:100%'><tr><td>"
				+ "<h3>Username</h3></td><td><input type='text' name='username' value='' class='input' required></input>"
				+ "</td></tr><tr><td>"
				+ "<h3>Password</h3></td><td><input type='password' name='password' value='' class='input' required></input>"
				+ "</td></tr><tr><td>"
				+ "<h3>First Name</h3></td><td><input type='text' name='firstname' value='' class='input' required></input>"
                + "</td></tr><tr><td>"
                + "<h3>Last Name</h3></td><td><input type='text' name='lastname' value='' class='input' required></input>"
				+ "</td></tr></table>" + "<br>");
				out.println("</form>" + "</div></div></div>");
				out.println("<button  data-toggle='modal' data-target='#myModal' class='btn btn-primary'>Create User </button></div>");
		out.println("<div id='myModal' class='modal fade' role='dialog'><div class='modal-dialog'><div class='modal-content'><div class='modal-header'> <button type='button' class='close' data-dismiss='modal'>&times;</button> <h4 class='modal-title'>Product Status</h4> </div>");  
		out.println("<div class='modal-body'><p>User Updated Successfully!.</p></div><div class='modal-footer'><button type='button' class='btn btn-default' data-dismiss='modal'>Close</button></div></div></div></div>");
			
		util.printHtml("Footer.html");
	}
}

File: smarthomes/src/NewCustomerPage.js
import './style.css';
import Header2 from './Header2';
import { useForm, Controller } from 'react-hook-form';
import userDetails from './UserDetails.json';
import {React, useState,useEffect} from 'react';

export default function NewCustomerPage(){
    const { register,control, handleSubmit, setError,setValue, formState: { errors } } = useForm();
    
    const [isFormVisible, setFormVisibility] = useState(true);
    const [updationMessage, setupdationMessage] = useState();

    useEffect(() => {
        // Set the default value for usertype when the component mounts
        setValue('usertype', 'customer');

        setValue('usertype', 'customer');
  }, [setValue]);

    const handleRegistration = (data) => {
        if (data.password !== data.repassword) {
            setError('repassword', { type: 'manual', message: "Passwords don't match" });
          } else {
            // You can send the form data to your server for registration and data storage here
            console.log('Form submitted', data);
            const userExists = userDetails.users.some(user => user.username === data.username);
            if (userExists) {
                setError('username', { type: 'manual', message: "Username already exists" });
            } else{
                userDetails.users.push({
                    username: data.username,
                    password: data.password,
                    usertype: data.usertype
                  });
                  setupdationMessage("Account created");
                  setFormVisibility(false);
               
            }
      
          }
      }
      const handleError = (errors) => {};

    const registerOptions = {
        username : {required : "Username cannot be blank"},
        password : { required : "Password cannot be blank" },
        repassword : {required : "Password cannot be blank"},
    };  
    return(
        <>
             <div className="Container">
                <Header2 />
                {isFormVisible && (
                <div className='post' style={{float: "none", width: "100%"}}>
                <h2 className='title meta'><a style={{fontSize: "24px"}}>Add New Customer</a></h2>
                <div className='entry'>
                <div style={{width:"400px", margin:"25px", marginLeft: "auto",marginRight: "auto"}}>
                
                <form onSubmit={handleSubmit(handleRegistration, handleError)}>
                    <table style={{width:'100%'}}><tr><td>
                    <h3>Username</h3></td><td><Controller
                    name='username'
                    control={control}
                    render={({ field }) => (
                        <>
                        <input type='text' name='username' className='input' {...register("username",registerOptions.username)} {...field} ></input>
                    <small style={{color:"red"}}>{errors?.username && errors.username.message}</small>
                    </>
                    )} />
                    
                    </td></tr><tr><td>
                    <h3>Password</h3></td><td><Controller
                    name='password'
                    control={control}
                    render={({ field }) => (
                    <>
                    <input type='password' name='password' className='input' {...register("password",registerOptions.password)} {...field}></input>
                    <small style={{color:"red"}}>{errors?.password && errors.password.message}</small>
                    </>
                    )}
                  />
                    </td></tr><tr><td>
                    <h3>Re-Password</h3></td><td><Controller
                    name='repassword'
                    control={control}
                    render={({ field }) => (
                    <>
                    <input type='password' name='repassword' className='input' {...register("repassword",registerOptions.repassword)} {...field}></input>
                    <small style={{color:"red"}}>{errors?.repassword     && errors.repassword.message}</small>
                    </>
                    )}
                    />
                    </td></tr><tr><td>
                    <h3>User Type</h3></td><td><Controller
                        name="usertype"
                        control={control}
                        render={({ field }) => (<select name='usertype' className='input' {...field}><option value='customer'>Customer</option><option value='storeManager'>Store Manager</option><option value='salesManager'>salesManager</option></select>
                        )}
                        />
                        </td></tr></table>
                    <input type='submit' className='btnbuy' name='ByUser' value='Create User' style={{float: "right",height: "20px", margin: "20px", marginRight: "10px"}}></input>
                </form>
                </div></div></div>
                )}
                {updationMessage && (
                <>
                    <div className='post' style={{float: "none", width: "100%"}}>
                   <div id='myModal' className='modal fade' role='dialog'><div className='modal-dialog'><div className='modal-content'><div className='modal-header'> <button type='button' className='close' data-dismiss='modal'>&times;</button> <h4 className='modal-title'>User Account Status</h4> </div>  
                  <div className='modal-body'><p>User Created Successfully!.</p></div><div className='modal-footer'><button type='button' className='btn btn-default' data-dismiss='modal'>Close</button></div></div></div></div>
                  <h4 style={{color:"red",align:"center"}}>User Created Successfully!</h4>
                  </div>
                  </> 
              )}
                
            </div>
        </>
    );
}